License file issues (#32)
r-devel/cran-cookbook: website/description_issues.qmd

### Details

Most acronyms that are not widely known should be explained. For example, you don’t have to explain *OLS*, *SEO*, or *DNA* but explanations should be provided for: *MEFM* or *OCCDS*. If you are unsure of the acronyms common knowledge, please document all acronyms in the package.

---

# LICENSE files

## Problem

> A file is referenced in the DESCRIPTION field for License when it's not required or the full license has been copied and submitted in the LICENSE file.

Reviewer: I am not a native English speaker, as many R users, and this sentence is a bit hard to understand to me. 

```suggestion
Some licenses do not need to be added verbatim, instead they are referenced in the DESCRIPTION file and the CRAN template is added to the package. You can find the template at the [CRAN website](https://cran.r-project.org/web/licenses/).
```

Perhaps also explain that if for other purposes they want to add the full text they can add it to the .Rbuildignore file so that when it is build and sent to CRAN the full license text won't be included.

## Solution

Remove the reference `+ file LICENSE` and delete the LICENSE file.

### Details

Most licenses don’t need an additional file as these are part of R. The list of which licenses that do require an additional file can be accessed within _R_ by searching for **Note: this is a template, needs + file LICENSE**:

```r

> path_licenses <- RShowDoc("license.db")

Reviewer: It might be simpler for users to use `list.files` equivalent:

```suggestion
path_licenses <- list.files(path = R.home(), "license.db", full.names = TRUE, recursive = TRUE)
```

licenses_doc <- as.data.frame(read.dcf(path_licenses))
```

> If an additional file is needed, it should only specify the author and year. 

Reviewer 2: This is not entirely true. Authors and year are needed for the MIT license. The BSD file license should contain three lines:

> YEAR:
> COPYRIGHT HOLDER:
> ORGANIZATION:

Author: i'll update this!

::: callout-tip
## Tip

> All licenses must be [FOSS](https://svn.r-project.org/R/trunk/share/licenses/license.db). 

Reviewer: There are (32) CRAN packages that aren't FOSS see ?available.packages:

```r
ap_foss <- available.packages(filters = "license/FOSS")
ap_null <- available.packages()
setdiff(rownames(ap_null), rownames(ap_foss))
```
Some of them were updated recently...

Reviewer 3: Some of these have licenses listed in `license.db` with `FOSS: no` (ACM, CC BY-NC(-SA)). ACM seems to be acceptable for historical packages only (`akima`, `tripack`). Others fail the `license/FOSS` filter because they transitively depend on packages with `FOSS: no` licenses. For example, `ggseas` is GPL-3 itself, but it depends on `seasonal`, which is also GPL-3, but in turn depends on `x13binary`. The latter has `file LICENSE` which automatically marks it non-FOSS despite [the LICENSE](https://cran.r-project.org/web/packages/x13binary/LICENSE) says

> This Software <...> is not subject to copyright in the United States

> The United States/Commerce hereby grant to User a royalty-free, nonexclusive
  license to use, copy, and create derivative works of the Software outside of
  the United States

...which seems to protect all four software freedoms from the FOSS definition.

Or is it non-FOSS because

> User agrees to make a good faith effort to use the Software in a way that does
  not cause damage, harm, or embarrassment to the United States/Commerce.

counts as a use restriction?

I am also curious about a [recent case of CC BY-NC-ND 4.0](https://stat.ethz.ch/pipermail/r-package-devel/2024q3/010949.html), which is listed in [license.db](https://svn.r-project.org/R/trunk/share/licenses/license.db) as non-FOSS. Unlike other non-FOSS CC licenses, there are no "NoDerivatives" packages on CRAN, and the authors eventually [changed the license to GPL-3](https://github.com/GfellerLab/SuperCell/commit/ea585a578101c3e83c817bc39d758b1345ce8c20) after at least one resubmission. Does the "NoDerivatives" clause prohibit CRAN from distributing binary packages?

Reviewer: I think that adding a restriction to not embarrass the United States is a restriction on what you can do with the code...

About non derivatives, we would probably need an authoritative source to confirm if CRAN/R is accepting that or considers binaries as derivative (This affects Bioconductor too). 

I think in the context of this PR, the sentence could be relaxed a little: 

```suggestion
Licenses should be FOSS, or allowed by [R](https://svn.r-project.org/R/trunk/share/licenses/license.db). 
```
